### PR TITLE
Fix train/test split overlap issue

### PR DESCRIPTION
**Pull request by aalok-sathe**
The original indices of `train_dataset` must be dropped *after* we use them to exclude train examples from the test set. Otherwise the current setting trivially excludes the *initial* `train_len` examples from test set, and among the remaining there can still be overlap.

### CODE REVIEW

**Comment by abhimishra91** [comment]
Thanks for that input. I will merge the request.
